src/boundaries/stagger_and_step.jl

```diff
@@ -0,0 +1,217 @@
+      from a uniform distribution in [-15, δ]. This mode fails
+      often but still manage to provide long enough stretch of
+      trajectories. 
+
+    * `:unif`: The next candidate is `x_c = x + u*r` with `r` 
+      taken from a uniform distribution [0,δ]. 
+    * `:adaptive`: The next candidate is `x_c = x + u*r` with 
+      `r` drawn from a gaussian distribution with variance  δ.
+      The variance changes according to a free parameter `f`  
+      such that `δ = δ/f` if no candidate is found and `δ = δ*f`
+      when it succeeds.    
+* `f = 1.1`: It is the free parameter for the adaptive stagger
+  method. 
+* `δ₀ = 1.`: This is the radius for the first stagger 
+trajectory search. The algorithm looks for a point 
+sufficiently close to the saddle before switching to the 
+stagger-and-step routine. The search radius must be large 
+enough to find a suitable initial. 
+[^Sweet2001]: D. Sweet, *et al.*, Phys. Rev. Lett. **86**, pp 2261  (2001)
+[^Sala2016]: M. Sala, *et al.*, Chaos **26**, pp 123124 (2016)
+"""
+function stagger_and_step!(ds::DynamicalSystem, x0, N::Int, isinside::Function; δ = 1e-10, Tm  = 30, 
+    f = 1.1, max_steps = Int(1e5), stagger_mode = :exp, δ₀ = 1.)
+
+    xi = stagger_trajectory!(ds, x0, Tm, isinside; δ₀, stagger_mode = :unif, max_steps) 
+    v = Vector{Vector{Float64}}(undef,N)
+    v[1] = xi
+@showprogress   for n in 1:N
+        if escape_time!(ds, xi, isinside) > Tm
+            reinit!(ds, xi; t0 = 0)
+        else
+            xp, Tp = get_stagger!(ds, xi, δ, Tm, isinside; stagger_mode, max_steps, f)
+            # The stagger step may fail. We reinitiate the algorithm from a new initial condition.
+            if Tp < 0
+                xp = stagger_trajectory!(ds, x0, Tm, isinside; δ₀, stagger_mode = :exp, max_steps, f) 
+                δ = 0.1
+            end
+            reinit!(ds, xp; t0 = 0)
+        end 
+        step!(ds)
+        xi = deepcopy(current_state(ds))
+        v[n] = xi
+    end
+    return v
+end
+
+
+    
+function escape_time!(ds, x0, isinside) 
+    x = deepcopy(x0) 
+    set_state!(ds,x)
+    reinit!(ds, x; t0 = 0)
+    k = 1; max_step = 10000;
+    while isinside(x) 
+        k > max_step && break
+        step!(ds)
+        x = current_state(ds)
+        k += 1
+    end
+    return current_time(ds)
+end
+
+function rand_u(δ, n; stagger_mode = :exp)
+    if stagger_mode == :exp 
+        a = -log10(δ)
+        s = (15-a)*rand() + a
+        u = (rand(n).- 0.5)
+        u = u/norm(u)
+        return u*10^-s
+    elseif stagger_mode == :unif
+        s = δ*rand()
+        u = (rand(n).- 0.5)
+        u = u/norm(u)
+        return u*s
+    elseif stagger_mode == :adaptive
+        s = δ*randn()
+        u = (rand(n).- 0.5)
+        u = u/norm(u)
+        return u*s
+    end
+end
+
+# This function searches a new candidate in a neighborhood of x0 
+# with a random search depending on some distribution. 
+# If the search fails it returns a negative time.
```

Comment:
Instead of having this as comments above the function, make it a documentation string.